test/fixtures/linter/autofix/snowflake/002_previously_parse_tree_damaging/after.sql

```diff
@@ -0,0 +1,14 @@
+MERGE INTO foo.bar AS tgt
+USING (
+  SELECT foo::DATE AS bar
+  FROM foo.bar
+  WHERE
+    split(foo, '|')[2] REGEXP '^\\d+\\-\\d+\\-\\d+ \\d+\\:\\d+$'
+    OR foo IN ('BAR', 'FOO')
+) AS src
+ON
+  src.foo = tgt.foo
+WHEN MATCHED THEN
+  UPDATE SET
+    tgt.foo = src.foo
+;
```

Comment:
@barrywhart  This test almost killed me. I think my V1 would have worked but for this.

The test checks that the parse tree is damaged, and given this damage that we apply no fixes.

this PR fixes the damaged Parse tree and cause this set of fixes to work, meaning that the test fails.

Perhaps a test that uses a dummy bad rule to break the tree would be better.

Reply:
I just switched it to its valid FIX output

Reply:
I'm sorry about the wild goose chase. I don't remember my thought process in creating this test, but clearly I should've documented my intent.

Reply:
No problem - its knowledge formed in the fire. I created an issue to try to help avoid this in the future